activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -438,18 +438,24 @@ def change_counter(key, amount, options)
           redis.then do |c|
             c = c.node_for(key) if c.is_a?(Redis::Distributed)
 
-            if options[:expires_in] && supports_expire_nx?
-              c.pipelined do |pipeline|
+            expires_in = options[:expires_in]
+
+            if expires_in && supports_expire_nx?
+              count, _ = c.pipelined do |pipeline|
+                pipeline.incrby(key, amount)
+                pipeline.call(:expire, key, expires_in.to_i, "NX")
+              end
+            elsif expires_in
```

Comment:
It's a bit of a nitpick, but I think it could be easier to read with a nested `if`:

```ruby
if expires_in
  if supports_expire_nx?
  else
  end
else

end
```